hibernate-core/src/main/java/org/hibernate/event/service/internal/EventListenerRegistryImpl.java

```diff
@@ -131,17 +128,15 @@ SessionFactoryImplementor getSessionFactory() {
 		return sessionFactory;
 	}
 
-	CallbackRegistryImpl getCallbackRegistry() {
+	CallbackRegistry getCallbackRegistry() {
 		return callbackRegistry;
 	}
 
 	@Override
 	public void prepare(MetadataImplementor metadata) {
 		if ( callbackBuilder == null ) {
 			// TODO : not needed anymore when the deprecate constructor will be removed
-			this.callbackBuilder = new CallbackBuilderLegacyImpl(
-					sessionFactory.getServiceRegistry().getService( ManagedBeanRegistry.class ),
-					metadata.getMetadataBuildingOptions().getReflectionManager()
+			this.callbackBuilder = CallbacksFactory.buildCallbackBuilder( sessionFactory, metadata.getMetadataBuildingOptions().getReflectionManager()
 			);
```

Comment:
Minor code style comment: let's not have this on a single line.

Reply:
ok